src/py42/exceptions.py

```diff
@@ -210,6 +210,17 @@ def __init__(self, exception, message=None):
         super(Py42BadRequestError, self).__init__(exception, message)
 
 
+class Py42BadRestoreError(Py42BadRequestError):
```

Comment:
Would it make sense to name this `Py42BadRestoreRequestError` instead for clarity that it's the request itself that was bad?